### DISCUSSION TITLE
Timestamp now() function

### DISCUSSION THREAD

**Discussion by ilinardic22** [google/cel-spec#339] [Q&A]
I would like to write an expression in which I check the time difference between the data I get in form of a Timestamp and the current moment. 

The expression would look like: `my_data_timestamp - time.now() < duration('1s')`

Is it possible to write an expression like that in CEL? I could not find a CEL function that returns the current timestamp value, but could it be done by writing a custom function? I am using CEL with Java.

**Comment by TristonianJones** [comment]
The CEL team prefers not to introduce a `now()` function since its behavior might imply side effects, or that calling now more than once would possibly produce a different value.
